graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -113,9 +110,11 @@ def __init__(self, class_name, kind, abstract, properties, class_fields):
             a SchemaElement with the given parameters
         """
         if kind == SchemaElement.ELEMENT_KIND_EDGE:
-            # Ensure that all edge classes have defined in/out properties.
-            _validate_edge_has_defined_endpoint_types(class_name, properties)
             _validate_edges_do_not_have_extra_links(class_name, properties)
+            if not abstract:
+                # Ensure that all non-abstract edge classes have defined in/out properties.
+                _validate_edge_has_defined_endpoint_types(class_name, properties)
```

Comment:
This fix is one that we identified in defacto. Basically, edges only need to have defined edge points if they are not abstract.